nodejs/init.sls

```diff
@@ -1,10 +1,24 @@
-node-pkgs:
-  pkg:
-    - installed
-    - names:
-      - npm
-      - nodejs-legacy
+{% import 'project/_vars.sls' as vars with context %}
+include:
+  - project.web.app
 
-# note: 'nodejs-legacy' is an ubuntu package that installs node then adds
-# a symlink to /usr/bin/node, which is where every other Linux distribution
-# puts the node executable.
+# Install NodeJS v4.x, to match local frontend dev setup
+nodejs_repo:
+  pkgrepo.managed:
+    - humanname: NodeSource Node.js PPA
+    - name: deb https://deb.nodesource.com/node_4.x {{ grains['lsb_distrib_codename'] }} main
+    - file: /etc/apt/sources.list.d/nodesource.list
+    - key_url: salt://project/web/nodesource.pub
+    - require_in:
+      - pkg: nodejs
+
+remove_nodejs_legacy:
+  pkg.removed:
+    - name: npm
+    - name: nodejs-legacy
+  required_in:
```

Comment:
this needs to be another list item under `pkg` and spelled `require_in`.  Or probably better to just make `nodejs` have a dependency on `pkg: remove_nodejs_legacy` since that's the order we more commonly write dependencies.